src/main/java/step2/domain/Judge.java

```diff
@@ -0,0 +1,47 @@
+package step2.domain;
+
+import step2.dto.JudgeResponse;
+import step2.dto.Lotto;
+import step2.dto.Price;
+import step2.dto.ShopResponse;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Judge {
+
+    private static final int MATCH_CASE = 6 + 1;
+
+    public JudgeResponse calculateResult(ShopResponse shopResponse, Lotto winningLotto) {
+        HashSet<Integer> winningNumbers = new HashSet<>(winningLotto.getNumbers());
+        int[] matchCount = new int[MATCH_CASE];
+        List<Integer> matches = shopResponse.getLottoList()
+                .stream()
+                .map(lotto -> calculateMatchCount(lotto, winningNumbers))
+                .collect(Collectors.toList());
+        matches.forEach(val -> matchCount[val]++);
+        double profitIndex = calculateMoneyWon(matchCount) / shopResponse.getOriginMoney();
+        return new JudgeResponse(matchCount[3], matchCount[4], matchCount[5], matchCount[6], profitIndex);
+    }
+
+    private double calculateMoneyWon(int[] matchCount) {
+        return Price.THREE_MATCH_PRICE * matchCount[3] +
+                Price.FOUR_MATCH_PRICE * matchCount[4] +
+                Price.FIVE_MATCH_PRICE * matchCount[5] +
+                Price.SIX_MATCH_PRICE * matchCount[6];
+    }
+
+    private int calculateMatchCount(Lotto lotto, HashSet<Integer> winningNumbers) {
+        int count = 0;
+        for (int idx = 0; idx < lotto.getNumbers().size(); idx++) {
+            count += oneIfNumberIsInHashSet(lotto.getNumbers().get(idx), winningNumbers);
+        }
```

Comment:
```suggestion
        for (int number: lotto.getNumbers()) {
            count += oneIfNumberIsInHashSet(number, winningNumbers);
        }
```

함수형 프로그래밍에 앞서, 일반적인 for 문에 대해서도 공부를 하시면 좋을 것 같습니다.
`lotto.getNumbers().get(idx)` 와 같은 표현을 저는 지양하는 편인데,
그 이유에 대해서는 [디미터 법칙](https://woowacourse.github.io/javable/post/2020-06-02-law-of-demeter/) 을 한번 공부해 보시면 좋을 것 같습니다.

Reply:
좋은 피드백 감사합니다ㅠㅠ